src/vs/base/parts/tree/browser/treeView.ts

```diff
@@ -1279,7 +1279,11 @@ export class TreeView extends HeightMap {
 		}
 
 		e.dataTransfer.effectAllowed = 'copyMove';
-		e.dataTransfer.setData(DataTransfers.RESOURCES, JSON.stringify([item.uri]));
+
+		// {{SQL CARBON EDIT}}
+		e.dataTransfer.setData(DataTransfers.RESOURCES, JSON.stringify([`${element.nodeTypeId}:${item.uri}?${element.metadata.schema ? element.metadata.schema + '.' + element.metadata.name : element.metadata.name}`]));
```

Comment:
I think instead of making changes in the `vs` side of code and then adding the `SQL Carbon Edit` tag, we should change these files instead - 

src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts
src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts

These files control the server tree. Changing the `treeview` in `vs` would change the behavior of all the trees